singer/src/main/java/com/pinterest/singer/utils/SingerUtils.java

```diff
@@ -78,7 +78,7 @@ public class SingerUtils {
 
   public static final FileSystem defaultFileSystem = FileSystems.getDefault();
   public static String HOSTNAME = getHostname();
-  public static List<String> HOSTNAME_PREFIXES = getHostnamePrefixes();
+  public static List<String> HOSTNAME_PREFIXES = getHostnamePrefixes("-");
```

Comment:
Where is the regex initially come from? I thought it would become a singer config.

Reply:
It does come from singer config, it gets applied when singer config is loaded, see:  https://github.com/pinterest/singer/pull/377/files#diff-cfc9cf543ef8be5e018e2c50a573b18f2268330cd83b68ca2bcde3c69ac3dae8R256